cs244b_project/src/lib.rs

```diff
@@ -515,6 +524,14 @@ impl StreamletInstance {
             >= (2.0 * (self.expected_peer_count + 1) as f64 / 3.0).ceil() as usize;
     }
 
+    fn should_vote(&mut self, message: &Message, voted_this_epoch: bool, epoch: u64, block: &Block, app_interface: &AppInterface) -> bool {
+        return !voted_this_epoch && 
+            self.check_from_leader(epoch, &message) &&
+            block.epoch == epoch &&
+            self.blockchain_manager.index_of_ancestor_chain(block.clone()).is_some() &&
+            app_interface.data_is_valid(&message);
```

Comment:
currently, this is just a dummy function that always returns true -- but making a case for our api!